build.gradle

```diff
@@ -67,6 +67,22 @@ task nativeHeaders {
     }
 }
 
+def writeNativeVersionSources = tasks.register("writeNativeVersionSources", WriteNativeVersionSources) {
+    ["main", "shared", "curses"].each {
+        nativeSources.from("src/${it}/cpp")
+        nativeSources.from("src/${it}/headers")
+    }
+
+    def generatedSources = project.layout.buildDirectory.dir("generated/sources")
+
+    nativeVersionHeaderFile = file("$nativeHeadersDir/native_platform_version.h")
```

Comment:
We should not put this into the same directory as the output of `nativeHeaders`: that's overlapping outputs!

Reply:
This is not overlapping outputs, since both tasks writing to the directory have an output file. The parent directory is not an output of any of the tasks.

Reply:
Ah, that's true. I think I'd still feel better if we had it in a separate directory.

Reply:
Both tasks now output to a separate directory.